src/main/java/seedu/findvisor/logic/parser/FindCommandParser.java

```diff
@@ -22,58 +29,92 @@
         }
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            argMultimap.verifyNoBlankPrefixValueFor(PREFIX_ADDRESS);
-            String addressKeyword = argMultimap.getValue(PREFIX_ADDRESS).get();
-            return new FindCommand(new PersonAddressPredicate(addressKeyword));
+            userInput = getUserInput(argMultimap, PREFIX_ADDRESS);
+            return new FindCommand(new PersonAddressPredicate(userInput));
+        }
+        if (argMultimap.getValue(PREFIX_MEETING).isPresent()) {
+            userInput = getUserInput(argMultimap, PREFIX_MEETING);
+            LocalDate meetingDate = verifyMeetingDateFormat(userInput);
+            return new FindCommand(new PersonMeetingPredicate(meetingDate));
         }
-
         argMultimap.verifyNoBlankPrefixValueFor(PREFIX_TAG);
         List<String> tagsKeywords = argMultimap.getAllValues(PREFIX_TAG);
         return new FindCommand(new PersonTagsPredicate(tagsKeywords));
     }
 
     /**
-     * Returns true if exactly one prefix is present in the given {@code ArgumentMultimap}.
+     * Returns true if exactly one prefix type is present in the given {@code ArgumentMultimap}.
      * @param argMultimap The {@link ArgumentMultimap} to check for the presence of prefixes.
      * @param prefixes A varargs array of {@link Prefix} objects to be checked in the {@code argMultimap}.
      * @return {@code true} if exactly one of the specified prefixes is present in the {@code ArgumentMultimap}.
      */
-    private boolean isSinglePrefixPresent(ArgumentMultimap argMultimap, Prefix... prefixes) {
+    private boolean isSinglePrefixTypePresent(ArgumentMultimap argMultimap, Prefix... prefixes) {
         long prefixCount = Stream.of(prefixes)
                 .filter(prefix -> argMultimap.getValue(prefix).isPresent())
                 .count();
         return prefixCount == 1;
     }
+
+    /**
+     * Retrieves the user input associated with a specific prefix. If specified prefix has a blank value,
+     * then a ParseException is thrown.
+     *
+     * @param argMultimap The {@link ArgumentMultimap} to get the specified prefix value.
+     * @param prefix The {@link Prefix} specified by the user.
+     * @return The user input string associated with the prefix.
+     * @throws ParseException If the prefix value is blank.
+     */
+    private String getUserInput(ArgumentMultimap argMultimap, Prefix prefix) throws ParseException {
+        argMultimap.verifyNoBlankPrefixValueFor(prefix);
+        return argMultimap.getValue(prefix).get();
+    }
+
+    /**
+     * Validates and parses the meeting date string to a {@code LocalDate} object.
+     *
+     * @param userInput The date string to parse.
+     * @return Parsed {@code LocalDate} object.
+     * @throws ParseException If the date string does not conform to the expected pattern {@code DATE_PATTERN}.
+     */
+    private LocalDate verifyMeetingDateFormat(String userInput) throws ParseException {
```

Comment:
Will `parseMeetingDate` be a better method name? Since you are returning the parsed LocalDate, not returning whether the date format is valid

Reply:
@Dethada Do you think it will be better to create a `parseMeetingDate` method in `ParserUtil.java`?

Reply:
I think that will be good